core/src/main/java/com/datastrato/gravitino/authorization/AccessControlManager.java

```diff
@@ -142,6 +164,8 @@ public boolean isServiceAdmin(String user) {
    * @return true, if the user is metalake admin, otherwise false.
    */
   public boolean isMetalakeAdmin(String user) {
-    return adminManager.isMetalakeAdmin(user);
+    synchronized (adminOperationLock) {
+      return adminManager.isMetalakeAdmin(user);
+    }
```

Comment:
I will suggest you create to help methods like `doWithAdminLock(xxx)` and `doWithNonAdminLock(xxx)` to make it more clear.

Reply:
Done.